killerbee/dev_template.py

```diff
@@ -0,0 +1,169 @@
+
+        if channel is None:
+            pass
+        else:
+            self.set_channel(channel, page)
+        
+        self.__stream_open = True
+
+    def sniffer_off(self) -> None:
+        self.__stream_open = False
+
+    def set_channel(self, channel: int, page: int=0) -> None:
+        self.capabilities.require(KBCapabilities.SETCHAN)
+
+        if self.handle is None:
+            raise Exception("Handle does not exist")
+
+        if channel >= 11 and channel <= 26:
+            self._channel = channel
+        else:
+            raise Exception('Invalid channel')
+
+    def inject(self, packet: bytes, channel: Optional[int]=None, count: int=1, delay: int=0, page: int=0) -> None:
+        self.capabilities.require(KBCapabilities.INJECT)
+
+        if self.handle is None:
+            raise Exception("Handle does not exist")
+
+
+        if len(packet) < 1:
+            raise Exception('Empty packet')
+        if len(packet) > 125:                   # 127 - 2 to accommodate FCS
+            raise Exception('Packet too long')
+
+        if channel is None:
+            pass
+        else:
+            self.set_channel(channel, page)
+
+    def pnext(self, timeout: int=100) -> Any:
+        self.capabilities.require(KBCapabilities.SNIFF)
+
+        if self.handle is None:
+            raise Exception("Handle does not exist")
+
+        if self.__stream_open == False:
+            self.sniffer_on()
+
+        packet: Optional[bytes] = None
+        frame: Optional[bytes] = None
+        validcrc: bool = False
+        rssi: Optional[int] = None
+
+        if frame is None:
+            pass
+        else:
+            if frame[-2:] == makeFCS(frame[:-2]):
+                validcrc = True
+
+        result: Dict[Union[int, str], Any] = {
+            0: frame, 
+            1: validcrc, 
+            2: rssi, 
+            'bytes': frame, 
+            'validcrc': validcrc, 
+            'rssi': rssi, 
+            'location': None,
+            'datetime': datetime.utcnow()
+        }
+        if rssi is None:
+            result['dbm'] = None
+        else:
+            result['dbm'] = rssi - 45
+
+        return result
+ 
+    def ping(self, da: Any, panid: Any, sa: Any, channel: Optional[int]=None, page: int=0) -> None:
+        raise Exception('Not yet implemented')
+
+    def jammer_on(self, channel: Optional[int]=None, page: int=0) -> None:
+        self.capabilities.require(KBCapabilities.PHYJAM_REFLEX)
+
+        if self.handle is None:
+            raise Exception("Handle does not exist")
+
+        if channel is None:
+            pass
+        else:
+            self.set_channel(channel, page)
+
+    def set_sync(self, sync: int=0xA70F) -> Any:
+        self.capabilities.require(KBCapabilities.SET_SYNC)
+
+        if self.handle is None:
+            raise Exception("Handle does not exist")
+
+    def jammer_off(self, channel: Optional[int]=None, page: int=0) -> None:
+        raise Exception('Not yet implemented')
```

Comment:
Maybe put in `raise NotImplementedError()` type stuff in a lot of these so people know they have to replace and don't leave stub too much?